notebooks/anonymization_one_hot_adult.ipynb

```diff
@@ -0,0 +1,303 @@
+     ]
+    }
+   ],
+   "source": [
+    "from sklearn.preprocessing import OneHotEncoder\n",
+    "import scipy\n",
+    "\n",
+    "preprocessor = OneHotEncoder(handle_unknown=\"ignore\")\n",
+    "\n",
+    "x_train = preprocessor.fit_transform(x_train)\n",
+    "x_test = preprocessor.transform(x_test)\n",
+    "if scipy.sparse.issparse(x_train):\n",
+    "    x_train = x_train.toarray().astype(int)\n",
+    "if scipy.sparse.issparse(x_test):\n",
+    "    x_test = x_test.toarray().astype(int)\n",
+    "\n",
+    "print(x_train)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Train decision tree model"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 23,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Base model accuracy:  0.814446287083103\n"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/Users/abigailt/Library/Python/3.9/lib/python/site-packages/sklearn/utils/deprecation.py:103: FutureWarning: The attribute `n_features_` is deprecated in 1.0 and will be removed in 1.2. Use `n_features_in_` instead.\n",
+      "  warnings.warn(msg, category=FutureWarning)\n"
+     ]
+    }
+   ],
+   "source": [
+    "from sklearn.tree import DecisionTreeClassifier\n",
+    "from art.estimators.classification.scikitlearn import ScikitlearnDecisionTreeClassifier\n",
+    "\n",
+    "model = DecisionTreeClassifier()\n",
+    "model.fit(x_train, y_train)\n",
+    "\n",
+    "art_classifier = ScikitlearnDecisionTreeClassifier(model)\n",
+    "\n",
+    "print('Base model accuracy: ', model.score(x_test, y_test))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Anonymize data\n",
+    "## k=100\n",
+    "\n",
+    "The data is anonymized on the quasi-identifiers: age, education-num, capital-gain, hours-per-week and with a privact parameter k=100.\n",
+    "\n",
+    "This means that each record in the anonymized dataset is identical to 99 others on the quasi-identifier values (i.e., when looking only at those features, the records are indistinguishable)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 25,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[[0 0 0 ... 0 1 0]\n",
+      " [0 0 0 ... 0 1 0]\n",
+      " [0 0 0 ... 0 1 0]\n",
+      " ...\n",
+      " [0 0 0 ... 0 1 0]\n",
+      " [0 0 0 ... 0 1 0]\n",
+      " [0 0 0 ... 0 1 0]]\n"
+     ]
+    }
+   ],
+   "source": [
+    "from apt.utils.datasets import ArrayDataset\n",
+    "from apt.anonymization import Anonymize\n",
+    "\n",
+    "x_train_predictions = np.array([np.argmax(arr) for arr in art_classifier.predict(x_train)])\n",
+    "\n",
+    "# QI = (race, sex)\n",
+    "QI = [53, 52, 51, 50, 49, 48, 47]\n",
+    "QI_slices = [[47, 48, 49, 50, 51], [52, 53]]\n",
```

Comment:
Maybe give an example with feature names too? Wouldn't it be the more common case?

Reply:
Not necessarily, plus I think looking at this example and others it would not be too difficult to figure it out.